Library/Homebrew/dev-cmd/update-license-data.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+require "commands"
+require "cli/parser"
+require "json"
+require "net/http"
+require "open-uri"
+
+module Homebrew
+  module_function
+
+  SPDX_FOLDER_PATH = (HOMEBREW_LIBRARY_PATH/"data").freeze
+  FILE_NAME = "spdx.json"
+  SPDX_DATA_URL = "https://raw.githubusercontent.com/spdx/license-list-data/HEAD/json/licenses.json"
+
+  def update_license_data_args
+    Homebrew::CLI::Parser.new do
+      usage_banner <<~EOS
+        `update_license_data` <cmd>
+
+          Update SPDX license data in the Homebrew repository.
+      EOS
+      switch "--fail-if-changed",
+             description: "Return a failing status code if current license data's version is different from"\
+                           "the upstream. This can be used to notify CI when the SPDX license data is out of date."
+
+      max_named 0
+    end
+  end
+
+  def update_license_data
+    update_license_data_args.parse
+    puts "Fetching newest version of SPDX License data..."
+    resp = Net::HTTP.get_response(URI.parse(SPDX_DATA_URL))
+
+    File.open(SPDX_FOLDER_PATH/FILE_NAME, "wb") do |file|
+      file.write(resp.body)
+    end
```

Comment:
```suggestion
    spdx_data = curl(SPDX_DATA_URL)
    SPDX_PATH.write(spdx_data)
```

This may not work as-is (will need testing) but the main points:
- use `curl` instead of `Net::HTTP` (to ensure proxies are used consistently)
- Write with `SPDX_PATH.write` (and use `SPDX_PATH.unlink` before if you need to delete)

Reply:
Hi @MikeMcQuaid , I am runnning into `Error: Will not overwrite spdx.json` 
On further investigation, I realise this is an expected behaviour from `Homebrew/extend/pathname.rb` that does not seem to allow any overwrite options. 
```  
def write(content, *open_args)
    raise "Will not overwrite #{self}" if exist?

    dirname.mkpath
    open("w", *open_args) { |f| f.write(content) }
  end
```

I can think of two options: 
1. We remove the file each time, before using `Pathname::write`
2. Modify the `write()` itself to take in any argument `allow_overwrite:bool` which defaults to `false` so it is compatible w existing callers. 

I think 1. Is a little cumbersome and not so elegant but there could be good reasons why `.write()` never allowed the option of overwriting.

Reply:
^ decided to go with (2)

Reply:
Ah I reread this and realised it's better to use `unlink()`. My bad, will modify things.